sdk/containers/azcontainerregistry/blob_client_test.go

```diff
@@ -8,99 +8,221 @@ package azcontainerregistry
+		require.NoError(t, recording.AddGeneralRegexSanitizer("00", hash, nil))
+	}
+	return blobDigest
 }
 
-func TestBlobClient_CancelUpload_fail(t *testing.T) {
-	startRecording(t)
-	endpoint, cred, options := getEndpointCredAndClientOptions(t)
-	ctx := context.Background()
-	client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
-	require.NoError(t, err)
-	_, err = client.CancelUpload(ctx, "wrong location", nil)
-	require.Error(t, err)
+func TestBlobClient(t *testing.T) {
+	repository, digest := buildImage(t)
+	blobDigest := blobDigest(t, repository, digest)
+
+	t.Run("CheckBlobExists", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		res, err := client.CheckBlobExists(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		require.Equal(t, blobDigest, *res.DockerContentDigest)
+	})
+
+	t.Run("CheckChunkExists", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		res, err := client.CheckChunkExists(ctx, repository, blobDigest, "bytes=0-299", nil)
+		require.NoError(t, err)
+		require.NotEmpty(t, *res.ContentLength)
+	})
+
+	t.Run("CheckChunkExists_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.CheckChunkExists(ctx, repository, "wrong digest", "bytes=0-299", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("CheckBlobExists_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.CheckBlobExists(ctx, repository, "wrong digest", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("CompleteUpload_wrongDigest", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		getRes, err := client.GetBlob(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		blob, err := io.ReadAll(getRes.BlobData)
+		require.NoError(t, err)
+		startRes, err := client.StartUpload(ctx, "hello-world", nil)
+		require.NoError(t, err)
+		uploadResp, err := client.uploadChunk(ctx, *startRes.Location, streaming.NopCloser(bytes.NewReader(blob)), nil)
+		require.NoError(t, err)
+		_, err = client.completeUpload(ctx, "sha256:00000000", *uploadResp.Location, nil)
+		require.Error(t, err)
```

Comment:
Is this error something we can test against (ie, one with an actual status code?). Or is it just text, and the user can't do much with it?

Reply:
In most cases (as in this test specifically) it's a `ResponseError`.